### PR TITLE
[Doc]update download docs for 2306 version[skip ci]

### PR DESCRIPTION
**Pull request by nvliyuan**
update the download page for 2306 release

Signed-off-by: liyuan <yuali@nvidia.com>

### CODE REVIEW

File: docs/download.md [hunk 1]
```diff
@@ -31,49 +31,53 @@ Software Requirements:
...
-* Support Delta Lake write with auto-optimization and auto-compaction on Databricks platforms
+* Deliver enhanced operator support with OOM retry framework to provide robust experience that minimizes OOM or GPU specific config changes across a wide variety of large scale workloads.
+* Deliver spill framework to reduce OOM issues to provide robust experience that minimizes OOM or GPU specific config changes across a wide variety of large scale workloads.
+* Deliver AQE skewed BHJ for all supported Databricks versions.
+* Support json to struct function on GPU.
+* Support StringTranslate function on GPU.
+* Support Windows function with str input in orderBy.
+* Support Regular expressions with line anchors in choice input.  
+* Support Rlike function with line anchor input.
+* Improve the performance of small ORC files reading.
```
Comment (sameerz) [inline docs/download.md:75]:
```suggestion
* Enhanced operator support with an OOM retry framework to minimize OOM or GPU specific config changes 
* Spill framework to reduce OOM issues to minimize OOM or GPU specific config changes 
* AQE for skewed broadcast hash join performance improvement
* Support JSON to struct
* Support StringTranslate 
* Support windows function with string input in order by clause
* Support regular expressions with line anchors in choice input  
* Support rlike function with line anchor input
* Improve the performance of ORC small file reads
```

File: docs/download.md [hunk 2]
```diff
@@ -31,49 +31,53 @@ Software Requirements:
...
 * Qualification and Profiling tool:
-  * Add support to recommend cluster shape options on Dataproc and EMR
-  * Add support for Databricks local mode with cost savings based on cluster metadata
-  * Add TCO calculator to estimate annualized cost savings, including estimated frequency for applications
-  * Add support in the qualification tool to generate estimated speed-up for ML functionality in Spark applications
+  * Qualification tool support Databricks Azure now.
+  * Improve Qualification tool to not require a live cluster, and only rely on READ permission.
+  * Improve Profiling tool to not require a live cluster, and only rely on READ permission.
+  * Improve Profiling tool recommendations to support more tuning options.
```
Comment (sameerz) [inline docs/download.md:80]:
```suggestion
  * Qualification tool support for Azure Databricks
  * The Qualification and Profiling tools do not require a live cluster, and only require read permissions on clusters
  * Improve Profiling tool recommendations to support more tuning options
```

File: docs/download.md [hunk 3]
```diff
@@ -31,49 +31,53 @@ Software Requirements:
...
 * Download the [RAPIDS
-  Accelerator for Apache Spark 23.04.1 jar](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/23.04.1/rapids-4-spark_2.12-23.04.1.jar)
+  Accelerator for Apache Spark 23.06.0 jar](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/23.06.0/rapids-4-spark_2.12-23.06.0.jar)
 
-This package is built against CUDA 11.8 and all CUDA 11.x versions are supported through [CUDA forward
+This package is built against CUDA 11.8, all CUDA 11.x and 12.x versions are supported through [CUDA forward
 compatibility](https://docs.nvidia.com/deploy/cuda-compatibility/index.html). It is tested
 on V100, T4, A2, A10, A30 and A100 GPUs with CUDA 11.0-11.5.  For those using other types of GPUs which
 do not have CUDA forward compatibility (for example, GeForce), CUDA 11.5 or later is required. Users will
 need to ensure the minimum driver (450.80.02) and CUDA toolkit are installed on each Spark node.
```
Comment (sameerz) [inline docs/download.md:52]:
```suggestion
compatibility](https://docs.nvidia.com/deploy/cuda-compatibility/index.html). It is tested
on V100, T4, A2, A10, A30, A100, L4 and H100  GPUs with CUDA 11.8-12.0.  For those using other types of GPUs 
which do not have CUDA forward compatibility (for example, GeForce), CUDA 11.8 or later is required. Users will
need to ensure the minimum driver (450.80.02) and CUDA toolkit are installed on each Spark node.
```

File: docs/download.md [hunk 4]
```diff
@@ -31,49 +31,52 @@ Software Requirements:
 
 	CUDA & NVIDIA Drivers*: 11.x & v450.80.02+
 
-	Apache Spark 3.1.1, 3.1.2, 3.1.3, 3.2.0, 3.2.1, 3.2.2, 3.2.3, 3.3.0, 3.3.1, 3.3.2, Databricks 10.4 ML LTS or 11.3 ML LTS Runtime and GCP Dataproc 2.0, Dataproc 2.1
+	Apache Spark 3.1.1, 3.1.2, 3.1.3, 3.2.0, 3.2.1, 3.2.2, 3.2.3, 3.2.4, 3.3.0, 3.3.1, 3.3.2, 3.3.3, 3.4.0 Databricks 10.4 ML LTS or 11.3 ML LTS Runtime and GCP Dataproc 2.0, Dataproc 2.1
```
Comment (sameerz) [inline docs/download.md:34]: Apologies for the late change - 3.3.3 is not released yet, my mistake.  
```suggestion
	Apache Spark 3.1.1, 3.1.2, 3.1.3, 3.2.0, 3.2.1, 3.2.2, 3.2.3, 3.2.4, 3.3.0, 3.3.1, 3.3.2, 3.4.0 Databricks 10.4 ML LTS or 11.3 ML LTS Runtime and GCP Dataproc 2.0, Dataproc 2.1
```

**Review by sameerz — APPROVED** [review]

**Comment by sameerz** [comment]
build

**Comment by nvliyuan** [comment]
build

**Pull request merged**